src/net/sourceforge/kolmafia/KoLmafia.java

```diff
@@ -1618,37 +1611,30 @@ public static final boolean checkRequirements(final List<AdventureResult> requir
 
   public static final boolean checkRequirements(
       final List<AdventureResult> requirements, final boolean retrieveItem) {
-    AdventureResult[] requirementsArray = new AdventureResult[requirements.size()];
-    requirements.toArray(requirementsArray);
-
     long actualCount = 0;
 
     // Check the items required for this quest,
     // retrieving any items which might be inside
     // of a closet somewhere.
 
-    for (int i = 0; i < requirementsArray.length; ++i) {
-      if (requirementsArray[i] == null) {
+    for (AdventureResult adventureResult : requirements) {
```

Comment:
I think this loop removes things from the list it's iterating over. That's probably why it was converted to an array first.

Please use iterators instead.

Reply:
Already did this before the review!